CMakeLists.txt

```diff
@@ -16,7 +16,7 @@
 #     will be removed as well as this note.
 #
 
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
This breaks compatibility with RHEL/CentOS 7, which carries CMake 2.8.12.2. For basic functionality (configuring and building the library and the unit tests), we try to stick to tooling that is [available in popular long-term support distributions](https://github.com/ARMmbed/mbedtls/issues/2896#issuecomment-603471273).

Updating to `cmake_minimum_required(VERSION 2.8.12.2)` would be ok as far as I know, although I think we should make this change only in development, not in our long-time support branches.

Reply:
@gilles-peskine-arm I suggest to contact the cmake developers https://gitlab.kitware.com/cmake/cmake about what makes most sense. But given that the cmake developers are good at providing new packages it might be good to go to 3.x soon.
And https://blog.kitware.com/cmake-2-8-12-2-available-for-download/ shows that it is 6,5 years old - and I guess within 6-12 months quite a few developers will use cmake 3.19 or later.
I fully understand the problem, and I just suggest that you make a good plan :)

Reply:
In #3801, I can see:
>CMake Deprecation Warning at CMakeLists.txt:19 (cmake_minimum_required):
  Compatibility with **CMake < 2.8.12** will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

Thus minimum version of 2.8.12 seems to be a good fit for RHEL/CentOS 7 and cmake 3.19, right?

Reply:
We don't want to constraint development tools too much. Support for CentOS as a development environment is a common request. How often CMake makes releases is irrelevant: what's relevant is how often CentOS and other distributions make releases. [CMake is keeping support for 2.8.12 for the same reason we want to keep support for 2.8.12](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4897#note_784038).

> 2.8.12 was included in some Linux distros' LTS branches and so is much more common than older 2.8 versions. That made it feel like a natural cutoff point, and I want to be conservative since this is the first time we're heading toward removing support for any policy's OLD behavior.

Reply:
As regards LTS branches, the versions prior to 2.18.2 are not removed yet (I am presuming this error is not fatal yet), just deprecated, so it kinda depends on how long we see the LTS branches living for. At some point we would have to backport this as otherwise the LTS branches may stop working with newer CMake.

Reply:
Thanx for the feedback @paul-elliott-arm @ronald-cron-arm and @gilles-peskine-arm - I have revised the PR to require 2.8.12 which I tested to be ok with cmake 3.19.0-rc1.

Just note that the mimimum cmake is not just a check of the used cmake version, it is also a load of rules - which have changed over time. Thus testing is golden....

Better?